lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -1982,7 +1982,7 @@ bool GenericRequirementsCheckListener::diagnoseUnsatisfiedRequirement(
 
 #pragma mark High-level entry points
 Type TypeChecker::typeCheckExpression(Expr *&expr, DeclContext *dc,
-                                      TypeLoc convertType,
+                                      Type convertType,
                                       ContextualTypePurpose convertTypePurpose,
                                       TypeCheckExprOptions options) {
   SolutionApplicationTarget target(
```

Comment:
How about removing the TypeLoc from SolutionApplicationTarget as well?

Reply:
ContextualFailures in the constraint system still use the TypeLoc (probably because its presence is substituting for a missing anchor). I can look into removing it, but it's probably going to be a lot more refactoring.